pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,136 @@
+package sso
+
+import (
+	"context"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/redhatsso"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services"
+	"github.com/golang/glog"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
+	"time"
+)
+
+type redhatssoService struct {
+	client redhatsso.SSOClient
+}
+
+func (r *redhatssoService) RegisterKafkaClientInSSO(accessToken string, kafkaNamespace string, orgId string) (string, *errors.ServiceError) {
+	// TODO: implement
+	return "", nil
+}
```

Comment:
@akoserwal It is not clear to me how we are going to implement this method with the new api

Reply:
KeycloakService 
```RegisterKafkaClientInSSO(kafkaNamespace string, orgId string) (string, *errors.ServiceError)```

I see the Implementation has a parameter to the method?
``` RegisterKafkaClientInSSO(accessToken string, kafkaNamespace string, orgId string)```

how this access token will be passed?

```suggestion
func (r *redhatssoService) RegisterKafkaClientInSSO(kafkaNamespace string, orgId string) (string, *errors.ServiceError) {
	accessToken, tokenErr := r.client.GetToken()
	svcData,err:=r.client.CreateServiceAccount(accessToken, kafkaClusterName, kafkaClusterName)
	if err != nil {
		return "", errors.NewWithCause(errors.ErrorFailedToCreateSSOClient, err, "failed to register Kafka Client in SSO")
	}
	if svcData.Secret != nil {
		return *svcData.Secret, nil
	}
	return "", errors.NewWithCause(errors.ErrorFailedToGetSSOClientSecret, secretErr, "failed to get sso client secret")
}
```

Reply:
All the method requires to have an access token and deals with errors getting it, so I used a proxy pattern to remove that logic for each method.

When you instantiate the service, it will return the proxy: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/c990a4316d557f4d7ac7c11cf2f5fcd32e9efbfe/pkg/services/sso/redhatsso.go

Reply:
make sense